src/pixelator/graph/backends/implementations.py

```diff
@@ -428,9 +517,104 @@ def __init__(
         """
         self._raw = raw
 
+    @staticmethod
+    def _build_plain_graph_from_edgelist(
+        df: pl.LazyFrame,
+        create_using: Union[nx.Graph, nx.MultiGraph],
+    ) -> Union[nx.Graph, nx.MultiGraph]:
+        g = nx.empty_graph(0, create_using)
+        g.add_edges_from(
+            (row[0], row[1])
+            for row in df.collect(streaming=True).iter_rows(
+                named=False, buffer_size=1000
+            )
+        )
+        return g
+
+    @staticmethod
+    def _build_graph_with_node_counts_from_edgelist(
+        df: pl.LazyFrame,
+        create_using: bool,
+    ) -> Union[nx.Graph, nx.MultiGraph]:
+        unique_markers = set(df.unique("marker").collect()["marker"].to_list())
+        initial_marker_dict = {marker: 0 for marker in unique_markers}
+
+        def _add_or_append_to_node(g, node, marker):
+            if g.nodes.get(node):
+                g.nodes.get(node)["markers"][marker] += 1
+            else:
+                marker_dict = initial_marker_dict.copy()
+                marker_dict[marker] += 1
+                g.add_node(node, markers=marker_dict)
+
+        g: nx.Graph = nx.empty_graph(0, create_using)
+
+        for row in (
+            df.select(["upia", "upib", "marker"])
+            .collect(streaming=True)
+            .iter_rows(named=False, buffer_size=1000)
+        ):
+            _add_or_append_to_node(g, row[0], row[2])
+            _add_or_append_to_node(g, row[1], row[2])
```

Comment:
I was also thinking that maybe inlining the `_add_or_append_to_node` nested function here could help a bit. Not sure though if it will matter that match to justify the hit in readability, maybe something for a quick benchmark?

Reply:
It seems to give a performance boost of roughly 10% in the microbenchmark - what do you think? Is it worth it?

Reply:
I think that is definitly worth it, the function is quite small so it remains quite imho.

Reply:
Ok. I've pushed it now. 😄